dvc/compare.py

```diff
@@ -222,6 +246,8 @@ def dropna(self, axis: str = "rows", how="any"):
 
         for n_row, row in enumerate(self):
             for n_col, col in enumerate(row):
+                if subset and self.keys()[n_col] not in subset:
```

Comment:
Can we change the loop order here? Brings the `subset` judgment to the outer loop? Besides, I guess we can just iterate over the cols in the subset?

Reply:
Not sure how we could invert the loop order and keep the functionality below.